include/deal.II/lac/lapack_support.h

```diff
@@ -22,6 +22,11 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+#ifdef DEAL_II_WITH_64BIT_BLAS
+typedef long long BlasIntType;
```

Comment:
I'd prefer if we put this into `namespace types`. It's ok to do this here rather than in `types.h`.

The spelling should then be `types::BlasInt`. Or maybe `types::BLASInt` or `types::BLAS_Int`? Can you check whether we typically capitalize BLAS in anything? We seem to do so for LAPACK.